public/docs/ts/latest/guide/dependency-injection.jade

```diff
@@ -683,61 +652,61 @@ code-example(format, language="html").
-// #enddocregion providers-factory-1
+
 +makeExample('dependency-injection/ts/app/heroes/hero.service.ts','internals', 'app/heroes/hero.service.ts (excerpt)')(format='.')
-// #docregion providers-factory-2
+
 :marked
   We can inject the `Logger`, but we can't inject the  boolean `isAuthorized`.
   We'll have to take over the creation of new instances of this `HeroService` with a factory provider.
 
   A factory provider needs a factory function:
-// #enddocregion providers-factory-2
+
 +makeExample('dependency-injection/ts/app/heroes/hero.service.provider.ts','factory', 'app/heroes/hero.service.provider.ts (excerpt)')(format='.')
-// #docregion providers-factory-3
+
 :marked
   Although the `HeroService` has no access to the `UserService`, our factory function does.
 
   We inject both the `Logger` and the `UserService` into the factory provider and let the injector pass them along to the factory function:
-// #enddocregion providers-factory-3
+
 +makeExample('dependency-injection/ts/app/heroes/hero.service.provider.ts','provider', 'app/heroes/hero.service.provider.ts (excerpt)')(format='.')
-// #docregion providers-factory-4
+
 .l-sub-section
   :marked
     The `useFactory` field tells Angular that the provider is a factory function
     whose implementation is the `heroServiceFactory`.
 
-    The `deps` property is an array of [provider tokens](#token).
+    The `deps` property is #{_an} #{_array} of [provider tokens](#token).
     The `Logger` and `UserService` classes serve as tokens for their own class providers.
     The injector resolves these tokens and injects the corresponding services into the matching factory function parameters.
-// #enddocregion providers-factory-4
-// #docregion providers-factory-5
-- var lang = current.path[1]
-- var anexportedvar = lang == 'dart' ? 'a constant' : 'an exported variable'
-- var variable = lang == 'dart' ? 'constant' : 'variable'
+
+- var exportedvar = _docsFor == 'dart' ? 'constant' : 'exported variable'
+- var variable = _docsFor == 'dart' ? 'constant' : 'variable'
 :marked
-  Notice that we captured the factory provider in #{anexportedvar}, `heroServiceProvider`.
+  Notice that we captured the factory provider in #{_an} #{exportedvar}, `heroServiceProvider`.
   This extra step makes the factory provider reusable.
   We can register our `HeroService` with this #{variable} wherever we need it.
 
   In our sample, we need it only in the `HeroesComponent`,
-  where it replaces the previous `HeroService` registration in the metadata `providers` array.
+  where it replaces the previous `HeroService` registration in the metadata `providers` #{_array}.
   Here we see the new and the old implementation side-by-side:
-// #enddocregion providers-factory-5
+
+- var stylePattern = { otl: /(providers.*),$/gm };
 +makeTabs(
   `dependency-injection/ts/app/heroes/heroes.component.ts,
   dependency-injection/ts/app/heroes/heroes.component.1.ts`,
```

Comment:
I get BAD FILENAME for v2 in the TS docs. (bad docregion?)

Also, the Dart docs show this:

```
import 'hero_list_component_2.dart';
import 'hero_service_1.dart';
/*
import 'hero_list_component.dart';
import 'hero_service.dart';
*/
```

They should instead look more like the v3 imports, shouldn't they?

```
import 'hero_list_component.dart';
import 'hero_service.dart';
```